clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -269,10 +269,10 @@ protected void onJoinComplete(int generation,
             this.joinedSubscription = newJoinedSubscription;
         }
 
-        // update the metadata and enforce a refresh to make sure the fetcher can start
-        // fetching data in the next iteration
+        // Update the metadata to include the full group subscription. The leader will trigger a rebalance
+        // if there are any metadata changes affecting any of the consumed partitions (whether or not this
+        // instance is subscribed to the topics).
         this.metadata.setTopics(subscriptions.groupSubscription());
-        if (!client.ensureFreshMetadata(Long.MAX_VALUE)) throw new TimeoutException();
```

Comment:
@hachikuji any brief comment about why this is not needed any more?

Reply:
There were a couple reasons. Primarily it seemed like an unnecessary optimization which added some complicated failures to think about. We have already updated the internal assignment state at this point, but if this call raises an exception, then the user may see this updated state prior to having their `onPartitionsAssigned` callback invoked. I was also happy to get rid of one of the final cases of indefinite blocking in the consumer.

Reply:
Sounds great! Thanks